provisioning/plugins/common/unsignedcorim_decoder.go

```diff
@@ -74,7 +91,7 @@ func UnsignedCorimDecoder(data []byte, xtr IExtractor) (*decoder.EndorsementDeco
 
 		if c.Triples.ReferenceValues != nil {
 			for _, rv := range *c.Triples.ReferenceValues {
-				swComp, err := xtr.SwCompExtractor(rv)
+				swComp, err := xtr.RefValExtractor(rv)
```

Comment:
should we call the variable `refVal` then?

Reply:
Yes, changed it. Though I admit it fixes the problem partially.
Longer term, we need to rename SwComponent in the .proto interface to more generic name. 
I will work on it in Part -2.